c7n/resources/ec2.py

```diff
@@ -838,6 +838,75 @@ def _run_instances_op(self, op, instance_ids):
                 raise
 
 
+@actions.register('reboot')
+class Reboot(BaseAction, StateTransitionFilter):
+    """reboots a previously running EC2 instance.
+
+    :Example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: ec2-reboot-instances
+            resource: ec2
+            query:
+              - instance-state-name: running
+            actions:
+              - reboot
+
+    http://docs.aws.amazon.com/cli/latest/reference/ec2/reboot-instances.html
+    """
+
+    valid_origin_states = ('running',)
+    schema = type_schema('reboot')
+    permissions = ('ec2:RebootInstances',)
+    batch_size = 10
+    exception = None
+
+    def _filter_ec2_with_volumes(self, instances):
+        return [i for i in instances if len(i['BlockDeviceMappings']) > 0]
+
+    def process(self, instances):
+        instances = self._filter_ec2_with_volumes(
+            self.filter_instance_state(instances))
+        if not len(instances):
+            return
+
+        client = utils.local_session(self.manager.session_factory).client('ec2')
+        failures = {}
+
+        # Play nice around aws having insufficient capacity...
```

Comment:
i'd assume we wouldn't need to the zone/type grouping for reboot on a started instance, since a started instance already has a capacity allocation. have you seen any issues here wrt to capacity errors?

Reply:
I have not seen capacity issues yet.  I was just using the ec2 stop code as an example.  Should I remove those bits?